### ISSUE TITLE
InPaint feather Falloff feature

### ISSUE THREAD

**Issue by RodTDai** [lllyasviel/Fooocus#1788]
Hey guys, please add a "feather" control for inPainting masks, it is very important because 90% of time inPainting on realistic  persons photos do not transition smoothly enough.
I tried to upload a custom mask with a blured mask, but it does not work, it clips the gradient and just use a dull mask.

Thanks

**Comment by mashb1t** [comment]
Do you mean a fade/fizzle in functionality to make inpainting not 100% hard on the edges?
Might relate to https://github.com/lllyasviel/Fooocus/issues/1443

**Comment by poor7** [comment]
Also I didn't find the "erase" (correction) tool there.

**Comment by RodTDai** [comment]
I don´t think it is related. anyway, feather mask feature is very important. I recall of having this on automatic1111 ages ago when i used.
Doing additional "photoshop" just to blend better a dull mask with the original image is not a really good approach. So a mask feather parameter should be super useful.
Thanks guys.

**Comment by thiner** [comment]
看不懂你的描述。。。看你的名字应该是懂中文的（姓戴？），你可以用中文描述你的需求，或者截个图说明什么是“feather mask”。

**Comment by RodTDai** [comment]
在Photoshop中，蒙版羽化是指模糊蒙版的边缘，创建蒙版区域和非蒙版区域之间的平滑过渡。

**Comment by thiner** [comment]
Maybe this option?
![image](https://github.com/lllyasviel/Fooocus/assets/1897227/03f8d8ee-1adb-48ee-8ccb-cbbaf628d1be)

**Comment by mashb1t** [comment]
Reference to https://github.com/lllyasviel/Fooocus/issues/1848#issue-2073564033 by @lanyusan:
> the new image generated with Fooocus with the mask would have sharp edge. I am wondering if there is any chance Fooocus could be enhanced to add inpainting edge softening or other techniques to address the problem.

**Comment by RodTDai** [comment]
> Maybe this option? ![image](https://private-user-images.githubusercontent.com/1897227/295192919-03f8d8ee-1adb-48ee-8ccb-cbbaf628d1be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwODM0NDUsIm5iZiI6MTcwNTA4MzE0NSwicGF0aCI6Ii8xODk3MjI3LzI5NTE5MjkxOS0wM2Y4ZDhlZS0xYWRiLTQ4ZWUtOGNjYi1jYmJhZjYyOGQxYmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTJUMTgxMjI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGQxOTgzMzBkYjYwMjNmOTRhY2VkMjg1Y2Q0NTcyYWFhZDAyZDEwODA2MjQ0MGFjMTA4NmU0NTg1OTViMjgzYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fM0lzSGnibLPTy8IgzmgcV9t0E3LlFDUBG732fTeP2w)

No, that is not mask blur/feathering
It refers to the softening of edges in a selection or mask, which creates a transition zone where the effect or adjustment blends gradually into the surrounding area. This is particularly useful for avoiding harsh, noticeable edges and achieving a more natural, integrated look

This is feather on a mask
![image](https://github.com/lllyasviel/Fooocus/assets/107771390/705d9412-a3fe-4d5a-bbff-84ec025362ce)

**Comment by RodTDai** [comment]
> I'm attempting to grasp the situation in this thread. As of now, we lack a mask feather feature, and it's not the "ControNet softness" that thiner mentioned. The absence of this feature is a significant drawback. Even if the model is unable to process gradient masks, a viable workaround could involve blending the original image with the inpainted area, utilizing a gradient soft mask in post-processing. It's essential to introduce a feather pixel size parameter in the UI to adjust the softening extent of the mask.
> Thanks guys
I'm attempting to grasp the situation in this thread. As of now, we lack a mask feather feature, and it's not the "ControNet softness" that thiner mentioned. The absence of this feature is a significant drawback. Even if the model is unable to process gradient masks, a viable workaround could involve blending the original image with the inpainted area, utilizing a gradient soft mask in post-processing. It's essential to introduce a feather pixel size parameter in the UI to adjust the softening extent of the mask.
Thanks guys

**Comment by mashb1t** [comment]
https://www.reddit.com/r/StableDiffusion/comments/1ankbwe/a1111_dev_a1111_forge_have_a_new_feature_called/

from https://github.com/lllyasviel/Fooocus/issues/2363#issue-2154914336
